wordpress-mobile/WordPress-Android
# [Modules Restructuring] Restructure Analytics Module

Parent #16562
Fixes #16563

This PR restructures the `analytics` module by disassociating the [WordPress-Analytics-Android](https://github.com/wordpress-mobile/WordPress-Analytics-Android) repo and removing the extra `WordPressAnalytics` level from it. This effectively makes `analytics` a pure module, without any extra inner levels or unnecessary build files lying around (like `settings.gradle`, `.gitignore`, etc).

PS: @zwarm I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since I have been seeing you handling [data coverage for WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/issues/16343) for the past few weeks/months, adding/updating track events,  etc. As such, I think you might be the best person to sign-off on that change for WPAndroid. 🥇 

-----

Question (❓): I wasn't sure about the `LICENCE` file, which point to `MIT` and as such I chose to keep it. The root level `LICENCE` that WPAndroid is using is `GPLv2`. Should we keep the additional `MIT` licence or remove it in favor of `GPLv2`? 🤔 

-----

To test:
- Build the app and install it on your device or emulator.
- Go to `App Settings` -> `Privacy settings` and enable `Collect information`.
- Open `Logcat` and verify that track events are being fired as expected (you can search by `🔵 Tracked: `).

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
